clients/drcachesim/tracer/tracer.cpp

```diff
@@ -1638,24 +1669,22 @@ event_post_syscall(void *drcontext, int sysnum)
     }
 
 #ifdef BUILD_PT_TRACER
-    if (!op_offline.get_value() || !op_enable_kernel_tracing.get_value())
-        return;
-    if (!is_in_tracing_mode(tracing_mode.load(std::memory_order_acquire)))
-        return;
-    if (!syscall_pt_trace_t::is_syscall_pt_trace_enabled(sysnum))
```

Comment:
Before we checked these before get_cur_recording_sysnum: are we sure it's safe to only call get_cur*?

Reply:
Yes, it will simply return INVALID_SYSNUM if any of these isn't true. I purposefully check get_cur_recording_sysnum here without checking those:
- we may not be in tracing mode now but may have been a while ago when we started pt tracing. We do not want to skip stopping the pt tracing in that case
- just in case we missed the post-syscall event for the syscall that started pt tracing, we want to catch that on the next syscall's post-syscall event (even if the next syscall isn't one where we enable syscall pt tracing).

I'll add to comments.

Reply:
Done.